packages/migrate/test/migrate.test.ts

```diff
@@ -1207,13 +1207,38 @@ export default class LocaleService extends Service {
         packageDir: project.baseDir,
         filesToMigrate: inputs,
         reporter,
+        mode: 'drain',
+      };
+
+      for await (const _ of migrate(input)) {
+        // no ops
+      }
+
+      expectFile(outputs[0]).toMatchSnapshot();
+    });
+
+    test('import type after infer it from hierarchy', async () => {
+      const [inputs, outputs] = prepareInputFiles(project, [
+        'basic.animal.ts',
+        'basic.food.ts',
+        'basic.index.ts',
+      ]);
+
+      const input: MigrateInput = {
+        projectRootDir: project.baseDir,
+        packageDir: project.baseDir,
+        filesToMigrate: inputs,
+        reporter,
+        mode: 'drain',
       };
 
       for await (const _ of migrate(input)) {
         // no ops
       }
 
       expectFile(outputs[0]).toMatchSnapshot();
```

Comment:
Add an assertion for the file that should have a type e.g. `feed(food: Food)` in the dog file assertion.

This will help in situations that we have snapshots updated without validating output.

Reply:
@egorio once this is resolved and validated in our consumer apps, this is good to merge

Reply:
Added. Will merge PR after validation.